### PR TITLE
Custom delimiter string

### PR DESCRIPTION
**Pull request by mohitkhullar**
Custom delimiter string.

### CODE REVIEW

File: tests/cdb2sql_options.test/runit [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        echo "failed $testname"
+        echo "see diffs here: $HOSTNAME"
+        echo "> diff -u ${PWD}/{$testname.expected,$testname.output}"
+        echo
+        exit 1
+    fi
+done
+
+echo
+exit 0
```
Comment (nirbhayc) [inline tests/cdb2sql_options.test/runit:31]: Use [compare_results.sh](https://github.com/bloomberg/comdb2/blob/master/tests/tools/compare_results.sh) instead?

Comment (mohitkhullar) [reply tests/cdb2sql_options.test/runit:31]: will need to change the script to start accepting additional options, the script already uses -d for database name.

Comment (nirbhayc) [reply tests/cdb2sql_options.test/runit:31]: You can put it in $CDB2_OPTIONS.

File: tools/cdb2sql/cdb2sql.cpp [hunk 2]
```diff
@@ -1699,12 +1735,13 @@ int main(int argc, char *argv[])
         {"cdb2cfg", required_argument, NULL, 'c'},
         {"file", required_argument, NULL, 'f'},
         {"gensql", required_argument, NULL, 'g'},
+        {"delim", required_argument, NULL, 'd'},
```
Comment (nirbhayc) [inline tools/cdb2sql/cdb2sql.cpp:1738]: Also add it to [usage_test]()https://github.com/bloomberg/comdb2/blob/master/tools/cdb2sql/cdb2sql.cpp#L127

File: tools/cdb2sql/cdb2sql.cpp [hunk 3]
```diff
@@ -1379,7 +1414,8 @@ static int run_statement(const char *sql, int ntypes, int *types,
         } else if (printmode & DISP_TABS) {
             fprintf(out, "\n");
         } else if (printmode & DISP_GENSQL) {
-            fprintf(out, ");\n");
+            fprintf(out, ");");
```
Comment (nirbhayc) [inline tools/cdb2sql/cdb2sql.cpp:1417]: We should not need `;` anymore.
`fprintf(out, ")");`

File: tools/cdb2sql/cdb2sql.cpp [hunk 4]
```diff
@@ -472,6 +477,20 @@ static bool skip_history(const char *line)
     return true;
 }
 
+int has_delimiter(char *line, int len, char *delimiter, int dlen)
+{
+    if (dlen > len)
+        return -1;
```
Comment (nirbhayc) [inline tools/cdb2sql/cdb2sql.cpp:483]: [optional] use bool instead - return false here.

File: tools/cdb2sql/cdb2sql.cpp (hunk 3 above)
Comment (mohitkhullar) [reply tools/cdb2sql/cdb2sql.cpp:1417]: This is to keep the current behavior, we use newline as delimiter but we output ';\n' in gensql.

File: tools/cdb2sql/cdb2sql.cpp (hunk 2 above)
Comment (mohitkhullar) [reply tools/cdb2sql/cdb2sql.cpp:1739]: done

File: tools/cdb2sql/cdb2sql.cpp (hunk 4 above)
Comment (mohitkhullar) [reply tools/cdb2sql/cdb2sql.cpp]: done

**Pull request merged**